SessionState/handler.ts

```diff
@@ -1,66 +1,77 @@
+import { readableReportSimplified } from "@pagopa/ts-commons/lib/reporters";
 import {
   IResponseErrorBadGateway,
   IResponseErrorForbiddenNotAuthorized,
   IResponseErrorGatewayTimeout,
   IResponseErrorInternal,
   IResponseSuccessJson,
   ResponseErrorBadGateway,
+  ResponseErrorForbiddenNotAuthorized,
   ResponseErrorGatewayTimeout,
   ResponseErrorInternal,
   ResponseSuccessJson,
   getResponseErrorForbiddenNotAuthorized
 } from "@pagopa/ts-commons/lib/responses";
+import { SequenceMiddleware } from "@pagopa/ts-commons/lib/sequence_middleware";
 import { defaultLog } from "@pagopa/winston-ts";
-import * as express from "express";
 
-import * as E from "fp-ts/Either";
-import * as TE from "fp-ts/TaskEither";
-
-import { readableReportSimplified } from "@pagopa/ts-commons/lib/reporters";
-import { flow, pipe } from "fp-ts/lib/function";
 import { IConfig } from "../utils/config";
-import { verifyUserEligibilityMiddleware } from "../utils/middlewares/user-eligibility-middleware";
-
-import { SessionState } from "../generated/definitions/external/SessionState";
-import { Client } from "../generated/definitions/fast-login/client";
 import { SpidLevel, gte } from "../utils/enums/SpidLevels";
+import { TokenTypes } from "../utils/enums/TokenTypes";
+import { verifyUserEligibilityMiddleware } from "../utils/middlewares/user-eligibility-middleware";
+import {
+  ExchangeJwtPayloadExtended,
+  exchangeJwtValidationMiddleware
+} from "../utils/middlewares/exchange-jwt-validation-middleware";
 import {
   HslJwtPayloadExtended,
   hslJwtValidationMiddleware
 } from "../utils/middlewares/hsl-jwt-validation-middleware";
 
+import { SessionState } from "../generated/definitions/external/SessionState";
+import { Client } from "../generated/definitions/fast-login/client";
+
 type SessionStateErrorResponsesT =
   | IResponseErrorForbiddenNotAuthorized
   | IResponseErrorInternal
   | IResponseErrorBadGateway
   | IResponseErrorGatewayTimeout;
 
 type SessionStateHandlerT = (
-  user: HslJwtPayloadExtended
+  user: HslJwtPayloadExtended | ExchangeJwtPayloadExtended
 ) => Promise<IResponseSuccessJson<SessionState> | SessionStateErrorResponsesT>;
 
 type SessionStateClient = Client<"ApiKeyAuth">;
 
-const canSeeProfile = (user: HslJwtPayloadExtended): boolean =>
-  HslJwtPayloadExtended.is(user) && gte(user.spid_level, SpidLevel.L2);
+const canSeeProfile = (
+  user: HslJwtPayloadExtended | ExchangeJwtPayloadExtended
+): boolean =>
+  (ExchangeJwtPayloadExtended.is(user) &&
+    user.token_type === TokenTypes.EXCHANGE) ||
+  (HslJwtPayloadExtended.is(user) && gte(user.spid_level, SpidLevel.L2));
 
 export const sessionStateHandler = (
   client: SessionStateClient
 ): SessionStateHandlerT => (
-  reqJwtPayload: HslJwtPayloadExtended
+  reqJwtPayload: HslJwtPayloadExtended | ExchangeJwtPayloadExtended
 ): ReturnType<SessionStateHandlerT> =>
   pipe(
     reqJwtPayload,
     TE.fromPredicate(
       user_data => canSeeProfile(user_data),
```

Comment:
We can use a more compact style
```suggestion
      canSeeProfile,
```